### PR TITLE
Needle: export NeedleResponse

### PR DESCRIPTION
**Pull request by Borewit**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

#### Changed existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: URL not applicable, `NeedleResponse` already defined
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/needle/needle-tests.ts [hunk 1]
```diff
@@ -78,10 +78,21 @@ function API_get() {
             // if no http:// is found, Needle will automagically prepend it.
         });
```
Comment (sindilevich) [inline types/needle/needle-tests.ts]:
```suggestion
let resp: needle.NeedleResponse;
```

File: types/needle/needle-tests.ts [hunk 2]
```diff
@@ -78,10 +78,21 @@ function API_get() {
...
 
+    // be able to type the response
+    function getResponse(): Promise<needle.NeedleResponse> {
```
Comment (sindilevich) [inline types/needle/needle-tests.ts]:
```suggestion
    resp = needle('get', 'google.com/search?q=syd+barrett');
```

**Review by nikeee — APPROVED** [review]

File: types/needle/needle-tests.ts [hunk 3]
```diff
@@ -78,10 +78,21 @@ function API_get() {
...
+        return needle('get', 'google.com/search?q=syd+barrett');
```
Comment (sindilevich) [inline types/needle/needle-tests.ts]:
```suggestion
        
```

File: types/needle/needle-tests.ts [hunk 4]
```diff
@@ -78,10 +78,21 @@ function API_get() {
...
+    }
```
Comment (sindilevich) [inline types/needle/needle-tests.ts]:
```suggestion
    
```

File: types/needle/needle-tests.ts [hunk 5]
```diff
@@ -78,10 +78,21 @@ function API_get() {
...
+
     // using callback
     needle.get('google.com/search?q=syd+barrett', (err, resp) => {
         // if no http:// is found, Needle will automagically prepend it.
     });
+
+    // assign response
+    let resp: needle.NeedleResponse;
```
Comment (sindilevich) [inline types/needle/needle-tests.ts]:
```suggestion
   
```

**Review by sindilevich — CHANGES_REQUESTED** [review]
A suggestion for a simpler code examples in the **needle-test.ts** file.

File: types/needle/needle-tests.ts (hunk 1 above)
Comment (Borewit) [reply types/needle/needle-tests.ts]: I already declared `let resp: needle.NeedleResponse` here: https://github.com/Borewit/DefinitelyTyped/blob/8ceafbecc7a4c2e0d23227b4c6387ee3fa70d75d/types/needle/needle-tests.ts#L92

Do you want me to do something with the return value?

Comment (sindilevich) [reply types/needle/needle-tests.ts]: I suggested moving the `resp`'s declaration here so you can demonstrate the `needle()` return type, in place of the whole `getResponse()` function.

You then will use `resp` at line 93.

**Review by sindilevich — APPROVED** [review]

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!

**Pull request merged**